front/components/actions/mcp/details/MCPToolOutputDetails.tsx

```diff
@@ -107,69 +103,23 @@ export function SqlQueryBlock({ resource }: SqlQueryBlockProps) {
 
 interface ToolGeneratedFileDetailsProps {
   resource: ToolGeneratedFileType;
-  icon: React.ComponentType<{ className?: string }>;
   owner: LightWorkspaceType;
 }
 
 export function ToolGeneratedFileDetails({
   resource,
-  icon,
   owner,
 }: ToolGeneratedFileDetailsProps) {
-  const sendNotification = useSendNotification();
-
-  const handleDownload = useCallback(() => {
-    try {
-      const downloadUrl = `/api/w/${owner.sId}/files/${resource.fileId}?action=download`;
-      // Open the download URL in a new tab/window. Otherwise we get a CORS error due to the redirection
-      // to cloud storage.
-      window.open(downloadUrl, "_blank");
-    } catch (error) {
-      console.error("Download failed:", error);
-      sendNotification({
-        title: "Download Failed",
-        type: "error",
-        description: "An error occurred while opening the download link.",
-      });
-    }
-  }, [resource.fileId, sendNotification, owner.sId]);
-
+  const file = {
+    ...resource,
+    sourceUrl: `/api/w/${owner.sId}/files/${resource.fileId}`,
+  };
   return (
-    <>
-      <div>
-        <Citation
-          className="w-48 min-w-48 max-w-48"
-          containerClassName="my-2"
-          onClick={handleDownload}
-          tooltip={resource.title}
-        >
-          <CitationIcons>
-            <Icon visual={icon} />
-          </CitationIcons>
-          <CitationTitle>{resource.title}</CitationTitle>
-        </Citation>
-      </div>
-      {resource.snippet && (
-        <CollapsibleComponent
-          rootProps={{ defaultOpen: false }}
-          triggerChildren={
-            <span className="text-sm font-semibold text-muted-foreground dark:text-muted-foreground-night">
-              Preview
-            </span>
-          }
-          contentChildren={
-            <div className="py-2">
-              <CodeBlock
-                className="language-csv max-h-60 overflow-y-auto"
-                wrapLongLines={true}
-              >
-                {resource.snippet}
-              </CodeBlock>
-            </div>
-          }
-        />
-      )}
-    </>
+    <AttachmentCitation
```

Comment:
second tldr; of the PR: we used the same component everywhere to display citations